src/adapter/src/catalog.rs

```diff
@@ -3837,13 +3908,25 @@ impl<S: Append> Catalog<S> {
         self.parse_item(create_sql, Some(&PlanContext::zero()))
     }
 
-    // Parses the given SQL string into a `CatalogItem`.
-    fn parse_item(
+    // Parses the given SQL string into a `CatalogItem`, using an explicit state
+    fn parse_item_state(
         &self,
+        state: &CatalogState,
```

Comment:
Same comment as above, it's a bit scary to be working with two different state's at the same time. It might be unavoidable, but I just wanted to at least point that out.

Reply:
Looking again at the code, I think it's possible. There are a couple of cases where `parse_item` currently uses `&self`, like in `self.resolve_storage_host_config(..)` when encountering a `CREATE SOURCE`. But, I don't actually need this case, I just need views, which are trivial. I'll try this a second